### PR TITLE
automate mobile EOL pages

### PR DESCRIPTION
**Pull request by sunnyzanchi**
this change refactors `getAgentName` to accept a filepath instead of
a subject, which should be less error-prone

## testing

- http://localhost:8000/docs/apm/agents/net-agent/getting-started/net-agent-eol-policy 
shouldn't have the .NET MAUI releases anymore (v0.0.1 and v0.0.2)
- http://localhost:8000/docs/mobile-monitoring/new-relic-mobile/get-started/mobile-agents-eol-policy 
should look the same as prod
- running `./scripts/releaseNotes.mjs` shouldn't include any mobile agents

### CODE REVIEW

File: src/components/EolPage.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ const releaseNotesQuery = graphql`
       }
     ) {
       nodes {
+        fileAbsolutePath
         frontmatter {
           releaseDate
           subject
```
Comment (clarkmcadoo) [inline src/components/EolPage.js:22]: `super nit`: we can get rid of subject since its not being used 🤷

**Comment by clarkmcadoo** [comment]
Looks great, we'll want to add this sweet update to jp/kr as well, then we're golden

File: src/components/EolPage.js (hunk 1 above)
Comment (sunnyzanchi) [reply src/components/EolPage.js:22]: ah good catch. will remove

File: src/i18n/content/jp/docs/mobile-monitoring/new-relic-mobile/get-started/mobile-agents-eol-policy.mdx [hunk 2]
```diff
@@ -26,190 +26,7 @@ iOS (以前の XCFramework)、Android、および React Native のモバイル
...
-      <td>
-        2021 年 8 月 23 日
-      </td>
-
-      <td>
-        2023 年 8 月 23 日
-      </td>
-    </tr>
-  </tbody>
+  <EolPage agent="ios" />
```
Comment (clarkmcadoo) [inline src/i18n/content/jp/docs/mobile-monitoring/new-relic-mobile/get-started/mobile-agents-eol-policy.mdx:29]: each of these are going to need their corresponding locales

Comment (sunnyzanchi) [reply src/i18n/content/jp/docs/mobile-monitoring/new-relic-mobile/get-started/mobile-agents-eol-policy.mdx]: ah i didn't see that before. just fixed!

File: src/i18n/content/kr/docs/mobile-monitoring/new-relic-mobile/get-started/mobile-agents-eol-policy.mdx [hunk 3]
```diff
@@ -26,190 +26,7 @@ translationType: machine
...
-      <td>
-        2021년 8월 23일
-      </td>
-
-      <td>
-        2023년 8월 23일
-      </td>
-    </tr>
-  </tbody>
+  <EolPage agent="ios" locale="jp" />
```
Comment (clarkmcadoo) [inline src/i18n/content/kr/docs/mobile-monitoring/new-relic-mobile/get-started/mobile-agents-eol-policy.mdx:29]: these need to be 'kr' or 'ko' for locale

Comment (sunnyzanchi) [reply src/i18n/content/kr/docs/mobile-monitoring/new-relic-mobile/get-started/mobile-agents-eol-policy.mdx:29]: 🤦 oops

**Review by clarkmcadoo — APPROVED** [review]

**Review by tabathadelane — APPROVED** [review]
.NET seems to be filtering as we expect! Both on EOL pages and the Release Note sections. LGTM

**Pull request merged**